### PR TITLE
gatekeeper: 3.12.0 -> 3.13.0

### PR DESCRIPTION
**Pull request by kashw2**
## Description of changes

change log: https://github.com/open-policy-agent/gatekeeper/releases/tag/v3.13.0

diff: https://github.com/open-policy-agent/gatekeeper/compare/v3.12.0...v3.13.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kashw2** [comment]
Result of `nixpkgs-review pr 249129` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gatekeeper</li>
  </ul>
</details>

File: pkgs/applications/networking/cluster/gatekeeper/default.nix [hunk 1]
```diff
@@ -2,17 +2,19 @@
...
   pname = "gatekeeper";
-  version = "3.12.0";
+  version = "3.13.0";
 
   src = fetchFromGitHub {
     owner = "open-policy-agent";
     repo = "gatekeeper";
     rev = "v${version}";
-    sha256 = "sha256-DGLod05gmDzItPOBaGc2X/oMXSzvH932Wa44R+DUt/U=";
+    sha256 = "sha256-7ciI8KgkC1NRYP8bFW5RnZ2oLXaCuPHQBd2tvgQ+YO8=";
```
Comment (onny) [inline pkgs/applications/networking/cluster/gatekeeper/default.nix:17]:
```suggestion
    hash = "sha256-7ciI8KgkC1NRYP8bFW5RnZ2oLXaCuPHQBd2tvgQ+YO8=";
```

File: pkgs/applications/networking/cluster/gatekeeper/default.nix [hunk 2]
```diff
@@ -2,17 +2,19 @@
 , buildGoModule
 , fetchFromGitHub
 , installShellFiles
+, gatekeeper
+, testers
```
Comment (onny) [inline pkgs/applications/networking/cluster/gatekeeper/default.nix:6]:
```suggestion
, testers
```
Where is this dependency used?

Comment (kashw2) [reply pkgs/applications/networking/cluster/gatekeeper/default.nix]: I'd originally planned to add passthru tests however had abandoned this idea and forgot to remove the `gatekeeper` and `testers` dependencies. 

fixed in: https://github.com/NixOS/nixpkgs/pull/249129/commits/229a5a31687fa48ef91d6ac67e2ef4faf7d5d9f9

File: pkgs/applications/networking/cluster/gatekeeper/default.nix (hunk 1 above)
Comment (kashw2) [reply pkgs/applications/networking/cluster/gatekeeper/default.nix]: fixed in https://github.com/NixOS/nixpkgs/pull/249129/commits/73f7d31627796709b9578f52cfa1548a954938c6

**Review by SuperSandro2000 — APPROVED** [review]
please squash into one commit

**Comment by kashw2** [comment]
Commits have been squashed into https://github.com/NixOS/nixpkgs/pull/249129/commits/ea3df1934f447a126a5a6235953e73b39458c674

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**